featurizers/fingerprints.py

```diff
@@ -20,88 +21,51 @@
 class MorganFingerprint(FingerprintTransformer):
     def __init__(
         self,
-        radius: int = 2,
-        n_bits: int = 2048,
-        use_chirality: bool = False,
-        use_bond_types: bool = True,
-        use_features: bool = False,
-        sparse: bool = False,
-        result_type: str = "default",
+        radius: int = 3,
+        includeChirality: bool = False,
+        useBondTypes: bool = True,
+        onlyNonzeroInvariants: bool = False,
+        includeRingMembership: bool = True,
+        countBounds: Optional[List] = None,
+        fpSize: int = 2048,
         n_jobs: int = 1,
+        sparse: bool = False,
+        count: bool = False,
     ):
-        assert result_type in ["default", "as_bit_vect", "hashed"]
+        FingerprintTransformer.__init__(self, n_jobs, sparse, count)
+
+        self.fp_generator_kwargs = {
+            "radius": radius,
+            "includeChirality": includeChirality,
+            "useBondTypes": useBondTypes,
+            "onlyNonzeroInvariants": onlyNonzeroInvariants,
+            "includeRingMembership": includeRingMembership,
+            "countBounds": countBounds,
+            "fpSize": fpSize,
+        }
```

Comment:
I just realized that this probably won't work. See [Scikit-learn tutorial on custom estimators](https://scikit-learn.org/stable/developers/develop.html#instantiation). In particular, this design will for 100% break cross-validation or Pipeline objects.

I think we're forced to have separate attributes, with Pythonic names. But maybe this won't be that bad - this way `_get_generator()` will be quite explicit.

BTW, we don't have a test for that, I think. We should add generic Scikit-learn tests of estimators, but maybe as a separate PR.

Reply:
Done. I've also added the scikit-lear estimator tests to our backlog